### PR TITLE
Simplifying the flow

### PR DESCRIPTION
**Pull request by ghost**
To the average programmer, looking at the original code shows plenty of branches that result in returning None. Because there are all contained in else statements, they can be somewhat hard to follow. For this reason, I put the bottom else statement out of the clause, and have it so that after each brach, if nothing returns, it would go to one return None statement.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by CodaFi** [comment]
Looks good!  Could you run clang-format really quickly to patch up the switch statement?

**Comment by pi1024e** [comment]
Did but clang format causes some other changes as well for some reason.

**Comment by owenv** [comment]
I recommend using the `git clang-format` tool. It will run `clang-format` on only the parts of the file you changed in a specific commit. Some parts of the compiler don't follow the style guidelines exactly but we want to avoid including unrelated style changes in PRs when possible

**Comment by pi1024e** [comment]
Done. thank you

File: lib/Basic/Platform.cpp [hunk 1]
```diff
@@ -195,21 +197,17 @@ StringRef swift::getPlatformNameForTriple(const llvm::Triple &triple) {
...
-      break;
-    case llvm::Triple::SubArchType::ARMSubArch_v7:
-      return "armv7";
-      break;
-    case llvm::Triple::SubArchType::ARMSubArch_v6:
-      return "armv6";
-      break;
+    if (Triple.isOSLinux()) {
+        switch (Triple.getSubArch()) {
+            case llvm::Triple::SubArchType::ARMSubArch_v7:
```
Comment (CodaFi) [inline lib/Basic/Platform.cpp]: This still isn't quite right.  LLVM style is to align the `case`s with the `switch`.  This whole block just needs to be outdented.

**Review by CodaFi — CHANGES_REQUESTED** [review]

**Comment by pi1024e** [comment]
Addressed

File: lib/Basic/Platform.cpp [hunk 2]
```diff
@@ -23,48 +23,47 @@ bool swift::tripleIsiOSSimulator(const llvm::Triple &triple) {
   return (triple.isiOS() &&
           // FIXME: transitional, this should eventually stop testing arch, and
           // switch to only checking the -environment field.
-          (triple.isSimulatorEnvironment() ||
-           arch == llvm::Triple::x86 || arch == llvm::Triple::x86_64));
+          (triple.isSimulatorEnvironment() || arch == llvm::Triple::x86 ||
```
Comment (xwu) [inline lib/Basic/Platform.cpp]: These line break changes here are unrelated to the commit; please clang-format only your own changes and restore this to the original version.

File: lib/Basic/Platform.cpp [hunk 3]
```diff
@@ -246,16 +241,16 @@ getArchForAppleTargetSpecificModuleTriple(const llvm::Triple &triple) {
...
-  //          .Case ("armv7k", "armv7k")
-  //          .Case ("armv7", "armv7")
-              .Default(tripleArchName);
+      .Cases("arm64", "aarch64", "arm64")
+      .Cases("x86_64", "amd64", "x86_64")
+      .Cases("i386", "i486", "i586", "i686", "i786", "i886", "i986", "i386")
+      .Cases("unknown", "", "unknown")
+      // These values are also supported, but are handled by the default case
+      // below:
+      //          .Case ("armv7s", "armv7s")
```
Comment (xwu) [inline lib/Basic/Platform.cpp]: Likewise here, please restore the original indentation as these are unrelated whitespace changes (and also cause the code in the comments to fall out of alignment with the lines above).

Comment (pi1024e) [reply lib/Basic/Platform.cpp]: I literally did that, and this is what happens every time.

**Comment by pi1024e** [comment]
I managed to fix everything

File: lib/Basic/Platform.cpp (hunk 2 above)
Comment (xwu) [reply lib/Basic/Platform.cpp]: This is not resolved. Please restore the original whitespace.

**Review by xwu — COMMENTED** [review]
See previous comments. Otherwise, seems fine to me.

**Comment by pi1024e** [comment]
Fixed and rebased!

File: lib/Basic/Platform.cpp [hunk 4]
```diff
@@ -32,8 +32,8 @@ bool swift::tripleIsAppleTVSimulator(const llvm::Triple &triple) {
   return (triple.isTvOS() &&
           // FIXME: transitional, this should eventually stop testing arch, and
           // switch to only checking the -environment field.
-          (triple.isSimulatorEnvironment() ||
-           arch == llvm::Triple::x86 || arch == llvm::Triple::x86_64));
+          (triple.isSimulatorEnvironment() || arch == llvm::Triple::x86 ||
```
Comment (xwu) [inline lib/Basic/Platform.cpp]: Still this straggling unrelated whitespace change.

**Comment by pi1024e** [comment]
Done

**Comment by CodaFi** [comment]
That looks better

@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Review by CodaFi — APPROVED** [review]
Thanks for sticking with us through the review!

**Comment by pi1024e** [comment]
No probs!

**Review by thejohnlima — APPROVED** [review]
